pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/executor/OpChainSchedulerService.java

```diff
@@ -100,6 +102,7 @@ public void runJob() {
             if (isFinished) {
               closeOpChain(operatorChain);
             } else if (thrown != null) {
+              // TODO: It would make sense to cancel OpChains if they returned an error-block.
```

Comment:
won't `isFinished` be set to true if opChain returns an error block?

Reply:
Yeah right now that's the case. In a follow-up I'll most likely change this behavior to:

```
finally {
  if (thrown != null || returnedErrorBlock) {
    cancelOpChain(operatorChain);
  } else if (isFinished) {
    closeOpChain(operatorChain);
  }
}
```